File: src/main/java/cn/cloudself/model/IntegerEntity.java
package cn.cloudself.model;

import javax.persistence.*;

/**
 * @author HerbLuo
 * @version 1.0.0.d
 *          <p>
 *          change logs:
 *          2017/5/16 HerbLuo 首次创建
 */
@Entity
public class IntegerEntity extends Number {

    private int id;

    @Id
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    @Transient
    public int getValue() {
        return id;
    }

    @Override
    public String toString() {
        return "" + id;
    }

    /**
     * Returns the value of the specified number as an {@code int},
     * which may involve rounding or truncation.
     *
     * @return the numeric value represented by this object after conversion
     * to type {@code int}.
     */
    @Override
    public int intValue() {
        return id;
    }

    /**
     * Returns the value of the specified number as a {@code long},
     * which may involve rounding or truncation.
     *
     * @return the numeric value represented by this object after conversion
     * to type {@code long}.
     */
    @Override
    public long longValue() {
        return id;
    }

    /**
     * Returns the value of the specified number as a {@code float},
     * which may involve rounding.
     *
     * @return the numeric value represented by this object after conversion
     * to type {@code float}.
     */
    @Override
    public float floatValue() {
        return id;
    }

    /**
     * Returns the value of the specified number as a {@code double},
     * which may involve rounding.
     *
     * @return the numeric value represented by this object after conversion
     * to type {@code double}.
     */
    @Override
    public double doubleValue() {
        return id;
    }
}

File: src/main/java/cn/cloudself/model/IntegerEntity.kt
package cn.cloudself.model

import javax.persistence.*

/**
 * @author HerbLuo
 * @version 1.0.0.d
 *
 *
 * change logs:
 * 2017/5/16 HerbLuo 首次创建
 */
@Entity
class IntegerEntity : Number() {

    @get:Id
    var id: Int = 0

    @Transient
    fun getValue(): Int {
        return id
    }

    override fun toString(): String {
        return "" + id
    }

    override fun toInt(): Int {
        return id
    }

    override fun toLong(): Long {
        return id.toLong()
    }

    override fun toFloat(): Float {
        return id.toFloat()
    }

    override fun toDouble(): Double {
        return id.toDouble()
    }

    override fun toShort(): Short {
        return id.toShort()
    }

    override fun toChar(): Char {
        return id.toChar()
    }

    override fun toByte(): Byte {
        return id.toByte()
    }
}
